src/main/java/pl/pandziak/ticketbookingapp/domain/reservation/ReservationInMemoryRepository.java

```diff
@@ -0,0 +1,41 @@
+package pl.pandziak.ticketbookingapp.domain.reservation;
+
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+final class ReservationInMemoryRepository implements ReservationRepository {
+
+    private final Map<String, Reservation> reservations = new ConcurrentHashMap<>();
+
+    @Override
+    public Reservation save(Reservation reservation) {
+        reservations.put(reservation.getId(), reservation);
+        return reservation;
+    }
+
+    @Override
+    public Optional<Reservation> findById(String id) {
+        return Optional.ofNullable(reservations.get(id));
+    }
+
+    @Override
+    public List<Reservation> findAll() {
+        return List.copyOf(reservations.values());
+    }
+
+    @Override
+    public void delete(Reservation reservation) {
+        reservations.remove(reservation.getId());
+    }
+
+    @Override
+    public void deleteAll() {
```

Comment:
Powiedzmy sobie szczerze - kiedy zamierzasz wywołać tę metodę?

Reply:
Tak, przyznaję się że zrobiłam tę metodę pod testy co nie jest prawidłowym rozwiązaniem. Przez to że repozytoria są już "in memory", nie tworzyłam na potrzeby testów drugiej, praktycznie takiej samej implementacji. Jakbym podpięła bazę sql lub nosql, ta metoda pewnie by się tutaj nie pojawiła.